kiota_abstractions/base_request_configuration.py

```diff
@@ -4,8 +4,9 @@
 # See License in the project root for license information.
 # ------------------------------------
 from dataclasses import dataclass
-from typing import Dict, List, Optional, Union
+from typing import Dict, List, Optional
 
+from .default_query_parameters import QueryParameters
 from .headers_collection import HeadersCollection
 from .request_option import RequestOption
```

Comment:
as far as I understand, we're not going to be generating request configuration classes for python anymore, correct?

In that case, shouldn't the query parameters be generic so people get proper auto completion?

if we do stop generating request configuration classes, I suggest the same derive, name, deprecate trick as query parameters to clean things up.

Reply:
> as far as I understand, we're not going to be generating request configuration classes for python anymore, correct?

That was the plan, in line with languages like dotnet. Is this a requirement or can we keep them for python?

> In that case, shouldn't the query parameters be generic so people get proper auto completion?

Currently it's just an empty class with no specific fields/parameters. Or is the issue naming?

> if we do stop generating request configuration classes, I suggest the same derive, name, deprecate trick as query parameters to clean things up.

Why deprecate it in abstractions? My understanding is that we deprecate the generated ones in favor of using the generic `RequestConfiguration` class in abstractions

Reply:
> That was the plan, in line with languages like dotnet. Is this a requirement or can we keep them for python?

The goal here is to reduce the generated code as much as possible without impacting the developer experience or the source compatibility. Anything we can avoid generating within those requirements, we should.

> Currently it's just an empty class with no specific fields/parameters. Or is the issue naming?

My concern here is that since we're not using generic parameters when they type the Python equivalent of
`client.users.get(o => o.QueryParameters.Select)`, they won't get auto-completion of Select here.

> Why deprecate it in abstractions?

My guess is that the currently generated code (without your recent changes) depends on BaseRequeestConfiguration. So removing it or changing it too much would be source breaking. But to accommodate for the generic query parameters type value, I'm suggesting to introduce a new class, and deprecate this one.

Hopefully all of this makes sense now?

Reply:
Gotcha. Deprecated `BaseRequestConfiguration` in favor of  the new `RequestConfiguration` class. Also added  `$filter`, `$select`, `$orderby`, `$count`, `$top`, `$skip` and $`expand` as generic parameters